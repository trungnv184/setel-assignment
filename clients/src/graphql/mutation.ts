import { gql } from '@apollo/client';

const ADD_PRODUCT_TO_CART = gql`
  mutation AddToCart($cartInput: CartInput) {
    addToCart(cartInput: $cartInput) {
      id
      name
      url
      price
    }
  }
`;

const CREATE_ORDER_PRODUCTS = gql`
  mutation CreateOrder($orderInput: OrderInput) {
    createOrder(orderInput: $orderInput) {
      orderItems {
        id
        name
        url
        price
        quantity
        createdDate
      }
      metadata {
        firstName
        lastName
        address
        phoneNumber
      }
      notes
    }
  }
`;

export { ADD_PRODUCT_TO_CART, CREATE_ORDER_PRODUCTS };
