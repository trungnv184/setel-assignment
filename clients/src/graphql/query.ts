import { gql } from '@apollo/client';

const GET_PRODUCTS = gql`
  query GetProducts {
    products {
      id
      name
      price
      url
      rating
    }
  }
`;

const GET_PRODUCTS_IN_CART = gql`
  query GetProductsCart {
    getProductsCart {
      id
      name
      price
      url
      quantity
    }
  }
`;

const GET_ORDER_BY_ID = gql`
  query GetOrder($orderId: String) {
    getOrder(orderId: $orderId) {
      id
      orderItems {
        id
        name
        price
        url
        quantity
      }
      metadata {
        firstName
        lastName
        address
        phoneNumber
      }
      notes
      state
      createdDate
      updatedDate
    }
  }
`;

const GET_ORDER_STATUS = gql`
  query GetOrderStatus($orderId: String) {
    getOrder(orderId: $orderId) {
      state
    }
  }
`;

export { GET_PRODUCTS, GET_PRODUCTS_IN_CART, GET_ORDER_BY_ID, GET_ORDER_STATUS };
