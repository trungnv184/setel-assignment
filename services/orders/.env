DELAY_DELIVERY = 60000
MONGO_URI = mongodb+srv://setel:3JQAQhzkjLuvjE9b@cluster0.9w2cg.mongodb.net/OrderPayments?retryWrites=true&w=majority