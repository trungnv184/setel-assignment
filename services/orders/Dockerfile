FROM node:12

WORKDIR /usr/src/app

COPY package*.json ./

RUN npm install

COPY . .

ENV MONGO_URI=mongodb+srv://setel:3JQAQhzkjLuvjE9b@cluster0.9w2cg.mongodb.net/OrderPayments?retryWrites=true&w=majority

ENV DELAY_DELIVERY=60000

CMD ["npm", "run", "start"]
