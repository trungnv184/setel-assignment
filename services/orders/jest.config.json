{
  "moduleDirectories": ["node_modules", "src"],
  "moduleFileExtensions": ["js", "json", "ts"],
  "testEnvironment": "node",
  "testRegex": ".spec.ts$",
  "transform": {
    "^.+\\.ts$": "ts-jest"
  },
  "coverageDirectory": "./coverage",
  "collectCoverage": true,
  "moduleNameMapper": {
    "^@schemas$": "<rootDir>/src/orders/schemas",
    "^@dto(.*)$": "<rootDir>/src/orders/dto$1",
    "^@configs(.*)$": "<rootDir>/src/orders/config$1",
    "^@common(.*)$": "<rootDir>/src/orders/common$1",
    "^@orders(.*)$": "<rootDir>/src/orders/$1",
    "^@utils(.*)$": "<rootDir>/src/orders/utils$1",
    "^@payments(.*)$": "<rootDir>/src/payments$1"
  }
}
