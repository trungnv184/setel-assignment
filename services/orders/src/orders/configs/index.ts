export const WAIT_FOR_PAYMENT_PROCESSING = 5000;
export const MONGO_URI =
  'mongodb+srv://setel:3JQAQhzkjLuvjE9b@cluster0.9w2cg.mongodb.net/OrderPayments?retryWrites=true&w=majority';
