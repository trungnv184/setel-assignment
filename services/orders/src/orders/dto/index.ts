export { CreateOrderDto } from './created-order.dto';
export { UpdatedOrderDto } from './updated-order.dto';
