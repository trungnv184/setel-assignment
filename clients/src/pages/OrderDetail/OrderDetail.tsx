import React from 'react';
import { ErrorMessage, OrderItems, OrderState, OrderSummary } from 'components';
import { useParams } from 'react-router-dom';
import { useQuery } from '@apollo/client';
import { GET_ORDER_BY_ID } from 'graphql/query';
import { Dimmer, Loader } from 'semantic-ui-react';
import { OrderResponse } from 'types/order-response';

const OrderDetail: React.FC<any> = () => {
  const { orderId } = useParams<{ orderId: string }>();
  const { data, loading, error } = useQuery(GET_ORDER_BY_ID, {
    variables: { orderId },
  });

  if (loading) {
    return (
      <Dimmer active={true}>
        <Loader />
      </Dimmer>
    );
  }

  if (error) return <ErrorMessage />;

  const orderResponse = data?.getOrder as OrderResponse;
  return (
    <>
      <OrderSummary metadata={orderResponse.metadata} notes={orderResponse.notes} />
      <OrderState />
      <OrderItems orderItems={orderResponse.orderItems} createdDate={orderResponse.createdDate} />
    </>
  );
};

export default OrderDetail;
