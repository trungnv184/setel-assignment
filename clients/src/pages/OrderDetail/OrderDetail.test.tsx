import { render, screen, waitFor } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { GET_ORDER_BY_ID } from 'graphql/query';
import { GET_ORDER_BY_ID_MOCK } from 'test/mocks/orders.mock';
import OrderDetail from './OrderDetail';

let isCalled = false;

const getMocks = () => {
  return [
    {
      request: {
        query: GET_ORDER_BY_ID,
      },
      result: () => {
        isCalled = true;
        return GET_ORDER_BY_ID_MOCK;
      },
    },
  ];
};
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => jest.fn().mockReturnValue({ orderId: '60cf5c724678b30a48521b09' }),
}));

beforeEach(() => {
  jest.resetAllMocks();
});

afterEach(() => {
  jest.clearAllMocks();
});

describe('OrderDetailPage', () => {
  test('Should render order detail page page', async () => {
    const { container } = render(
      <MockedProvider mocks={getMocks()} addTypename={false}>
        <OrderDetail />
      </MockedProvider>
    );

    await waitFor(() => {
      expect(isCalled).toBeTruthy();
    });

    expect(screen.getByText(/Trung/i)).toBeInTheDocument();
    expect(screen.getByText(/Nguyen/i)).toBeInTheDocument();
    expect(screen.getByText(/20-06-2021/)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
