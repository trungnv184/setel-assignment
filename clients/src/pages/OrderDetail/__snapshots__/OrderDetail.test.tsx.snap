// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetailPage Should render order detail page page 1`] = `
<div>
  <div
    class="ui segment"
  >
    <div
      class="ui fluid left aligned container"
    >
      <h3
        class="ui header"
      >
        # Order Summary
      </h3>
      <div
        class="sc-hKFxyN idPHKU"
      >
        <span
          class="title"
        >
          First Name:
        </span>
        <span
          class="content"
        >
          Trung
        </span>
      </div>
      <div
        class="sc-hKFxyN idPHKU"
      >
        <span
          class="title"
        >
          Last Name:
        </span>
        <span
          class="content"
        >
          Nguyen
        </span>
      </div>
      <div
        class="sc-hKFxyN idPHKU"
      >
        <span
          class="title"
        >
          Address:
        </span>
        <span
          class="content"
        >
          HCM
        </span>
      </div>
      <div
        class="sc-hKFxyN idPHKU"
      >
        <span
          class="title"
        >
          Phone Number:
        </span>
        <span
          class="content"
        >
          0906925896
        </span>
      </div>
      <div
        class="sc-hKFxyN idPHKU"
      >
        <span
          class="title"
        >
          Notes:
        </span>
        <span
          class="content"
        >
          Nothing
        </span>
      </div>
    </div>
  </div>
  <div
    class="ui segment"
  >
    <h3
      class="ui header"
    >
      Order Status
    </h3>
    <div
      class="ui ordered steps"
    >
      <div
        class="completed step"
      >
        <div
          class="content"
        >
          <div
            class="title"
          >
            CREATED
          </div>
          <div
            class="description"
          >
            Your order have been created
          </div>
        </div>
      </div>
      <div
        class="active completed step"
      >
        <div
          class="content"
        >
          <div
            class="title"
          >
            CONFIRMED
          </div>
          <div
            class="description"
          >
            Your order have been confirmed
          </div>
        </div>
      </div>
      <div
        class="active completed step"
      >
        <div
          class="content"
        >
          <div
            class="title"
          >
            DELIVERED
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ui segment"
  >
    <div
      class="ui green label"
    >
      <i
        aria-hidden="true"
        class="book icon"
      />
       
      Selected Books 1
    </div>
    <div
      class="ui blue label"
    >
      <i
        aria-hidden="true"
        class="payment icon"
      />
       
      500 ₫
    </div>
    <div
      class="ui purple label"
    >
      <i
        aria-hidden="true"
        class="calendar icon"
      />
       
      20-06-2021
    </div>
    <div
      class="ui six column grid"
    >
      <div
        class="column"
      >
        <div
          class="sc-gtsrHT igIUx"
        >
          <img
            class="ui small image"
            src="url"
          />
          <div
            class="sc-dlnjwi guJDWs"
            data-testid="book-name"
          >
            Book
          </div>
          <div
            class="sc-dlnjwi guJDWs"
          >
            Price:
            500 ₫
          </div>
          <div
            class="sc-dlnjwi dHdVQU"
          >
            Quantity:
            1
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
